base_layer/core/src/mempool/service/inbound_handlers.rs

```diff
@@ -28,22 +28,34 @@ use crate::{
     },
 };
 use std::sync::Arc;
-use tari_transactions::transaction::Transaction;
+
+use tari_comms::types::CommsPublicKey;
+use tari_comms_dht::{
+    domain_message::OutboundDomainMessage,
+    envelope::NodeDestination,
+    outbound::{OutboundEncryption, OutboundMessageRequester},
+};
+use tari_p2p::tari_message::TariMessageType;
+use tari_transactions::{proto::types::Transaction as ProtoTransaction, transaction::Transaction};
 
 /// The MempoolInboundHandlers is used to handle all received inbound mempool requests and transactions from remote
 /// nodes.
 pub struct MempoolInboundHandlers<T>
 where T: BlockchainBackend
 {
     mempool: Mempool<T>,
+    outbound_message_service: OutboundMessageRequester,
```

Comment:
I think it might be worthwhile to add a "propagate_transaction" function to the outbound interface of the MempoolService and rather than give the inbound handler access to the OutboundMessageRequester it makes use of the outbound interface to broadcast the transaction.